File: FSharp/HelloWorld/Lib/K01.fs
namespace HelloWorld.Lib

type K01() =
    interface IQuestion with
        member this.Question1() = printfn "Hello World!　ようこそF#言語の世界へ！"
        
        member this.Question2() =
            printfn "Hello World!"
            printfn "ようこそ"
            printfn "F#言語の世界へ！"
            
        member this.Question3() =
            printfn $"整数：{12345}"
            printfn $"実数：{123.456789}"
            printfn $"文字：{'A'}"
            printfn "文字列：%s" "ABCdef"
            
        member this.Question4() =
            printfn "              ##";
            printfn "             #  #";
            printfn "             #  #";
            printfn "            #    #";
            printfn "           #      #";
            printfn "         ##        ##";
            printfn "       ##            ##";
            printfn "    ###                ###";
            printfn " ###       ##    ##       ###";
            printfn "##        #  #  #  #        ##";
            printfn "##         ##    ##         ##";
            printfn " ##     #            #     ##";
            printfn "  ###     ##########     ###";
            printfn "     ###              ###";
            printfn "        ##############"

File: Haskell/HelloWorld/src/K01.hs
module K01 (K01(..)) where

import Prelude
import Question (Question(..))

data K01 = K01

instance Question K01 where
  question1 _ = putStrLn "Hello World!　ようこそHaskell言語の世界へ！"
  question2 _ = do
    putStrLn "Hello World!"
    putStrLn "ようこそ"
    putStrLn "Haskell言語の世界へ！"
  question3 _ = do
    putStrLn $ "整数：" <> show 12345
    putStrLn $ "実数：" <> show 123.456789
    putStrLn $ "文字：" <> "A"
    putStrLn $ "文字列：" <> "ABCdef"
  question4 _ = do
    putStrLn "              ##"
    putStrLn "             #  #"
    putStrLn "             #  #"
    putStrLn "            #    #"
    putStrLn "           #      #"
    putStrLn "         ##        ##"
    putStrLn "       ##            ##"
    putStrLn "    ###                ###"
    putStrLn " ###       ##    ##       ###"
    putStrLn "##        #  #  #  #        ##"
    putStrLn "##         ##    ##         ##"
    putStrLn " ##     #            #     ##"
    putStrLn "  ###     ##########     ###"
    putStrLn "     ###              ###"
    putStrLn "        ##############"